src/linear_relation/canonical.rs

```diff
@@ -163,6 +163,17 @@ impl<G: PrimeGroup> CanonicalLinearRelation<G> {
         }
 
         // Only include constraints that are non-trivial (not zero constraints)
+        // QUESTION: Should this actually be done? In the 0 = [] case, this seems to be no loss. In
+        // the error case, this precludes including an always-false OR branch.
```

Comment:
my understanding is that we do not need trivially false OR branches, so I'd be happy just removing the above comment